content/en/blog/_posts/2023-04-14-1.27-updates-on-speeding-up-pod-startup.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: blog
+title: "1.27 updates on speeding up Pod startup"
+date: 2023-04-14T00:00:00+0000
+slug: speed-up-pod-startup
+---
+
+**Authors**: Paco Xu(DaoCloud), Sergey Kanzhelev(Google), Ruiwen Zhao(Google)
+
+How can Pod start-up be accelerated on nodes in large clusters? This is a common issue that
+cluster administrators may face.
+
+This blog post focuses on methods to speed up pod start-up from the kubelet side. It does not
+involve the creation time of pods by controller-manager through kube-apiserver, nor does it
+include scheduling time for pods or webhooks executed on it.
+
+We have mentioned some important factors here to consider from the kubelet's perspective, but
+this is not an exhaustive list. As Kubernetes v1.27 is set to release soon, this blog highlights
+significant changes in v1.27 that aid in speeding up pod start-up.
+
+## Parallel Image Pulls
+
+Pulling images always takes some time and what's worse is that image pulls are done serially by
+default. In other words, kubelet will send only one image pull request to the image service at
+a time. Other image pull requests have to wait until the one being processed is complete.
+
+To enable parallel image pulls, set the `serializeImagePulls` field to false in the kubelet
+configuration. When `serializeImagePulls` is disabled, requests for image pulls are immediately
+sent to the image service and multiple images can be pulled concurrently.
+
+### Maximum parallel image pulls will help secure your node from overloading on image pulling
+
+We are introducing a new feature in kubelet that sets a limit on the number of parallel image
+pulls at the node level. This limit restricts the maximum number of images that can be pulled
+simultaneously. If there is an image pull request beyond this limit, it will be blocked until
+one of the ongoing image pulls finishes. Before enabling this feature, please ensure that your
+container runtime's image service can handle parallel image pulls effectively.
+
+To limit the number of simultaneous image pulls, you can configure the `maxParallelImagePulls`
+field in kubelet. By setting `maxParallelImagePulls` to a value of _n_, only _n_ images will
+be pulled concurrently. Any additional image pulls beyond this limit will wait until at least
+one ongoing pull is complete.
+
+You can find more details in the associated KEP: [Kubelet limit of Parallel Image Pulls](https://kep.k8s.io/3673)
+ (KEP-3673). The improvement will arrive as part of Kubernetes v1.27 (scheduled for release in April 2023).
```

Comment:
You'll need to reword this now; v1.27 is released. Ideally, also link to the docs for managing image pull parallelism.